internal/hcsoci/resources_lcow.go

```diff
@@ -79,18 +80,25 @@ func allocateLinuxResources(ctx context.Context, coi *createOptionsInternal, res
 			l := log.G(ctx).WithField("mount", fmt.Sprintf("%+v", mount))
 			if mount.Type == "physical-disk" {
 				l.Debug("hcsshim::allocateLinuxResources Hot-adding SCSI physical disk for OCI mount")
-				_, _, err := coi.HostingSystem.AddSCSIPhysicalDisk(ctx, hostPath, uvmPathForShare, readOnly)
+				_, _, _, err := coi.HostingSystem.AddSCSIPhysicalDisk(ctx, hostPath, uvmPathForShare, readOnly)
 				if err != nil {
 					return fmt.Errorf("adding SCSI physical disk mount %+v: %s", mount, err)
 				}
 				resources.scsiMounts = append(resources.scsiMounts, scsiMount{path: hostPath})
 				coi.Spec.Mounts[i].Type = "none"
 			} else if mount.Type == "virtual-disk" || mount.Type == "automanage-virtual-disk" {
 				l.Debug("hcsshim::allocateLinuxResources Hot-adding SCSI virtual disk for OCI mount")
-				_, _, err := coi.HostingSystem.AddSCSI(ctx, hostPath, uvmPathForShare, readOnly)
+				uvmPathForShare = fmt.Sprintf(lcowGlobalMountPrefix, i)
```

Comment:
You need this logic for physical disk above as well

Reply:
@abwah was this resolved?

Reply:
yes, updated

Reply:
Do we not also need to update uvmPathForFile?

Reply:
Good catch. Fixed